From e81d23480a635c9f3e02081b6e6a70c04217454f Mon Sep 17 00:00:00 2001
From: Nickey Yang <nickey.yang@rock-chips.com>
Date: Mon, 3 Aug 2020 17:24:17 +0800
Subject: [PATCH] ARM: dts: rv1126: add evb v13 board support

v13 board solve the problem of insufficient amplitude
of usb in low-speed handshake phase by usb_dm pull up
and it control by gpio0_c1.
gpio0_c1 and pwm3 are one pin, so pwm3 used for
backlight is now provided by pwm0.

Signed-off-by: Nickey Yang <nickey.yang@rock-chips.com>
Change-Id: Ia4169f12217c382e5389060746be763b6e7ff4c3
---
 arch/arm/boot/dts/Makefile                |  1 +
 arch/arm/boot/dts/rv1126-evb-ddr3-v13.dts | 31 +++++++++++++++++++++++
 2 files changed, 32 insertions(+)
 create mode 100644 arch/arm/boot/dts/rv1126-evb-ddr3-v13.dts

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index 4f04cf7ddb27..4c7159c9f105 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -860,6 +860,7 @@ dtb-$(CONFIG_ARCH_ROCKCHIP) += \
 	rv1126-evb-ddr3-v12-spi-nand.dtb \
 	rv1126-evb-ddr3-v12-spi-nor.dtb \
 	rv1126-evb-ddr3-v12-tb-emmc.dtb \
+	rv1126-evb-ddr3-v13.dtb \
 	rv1126-evb-lp3-v10.dtb \
 	rv1126-iotest-v10.dtb \
 	rv1126-ipc2-ddr3-v10.dtb \
diff --git a/arch/arm/boot/dts/rv1126-evb-ddr3-v13.dts b/arch/arm/boot/dts/rv1126-evb-ddr3-v13.dts
new file mode 100644
index 000000000000..4e2bec9abf00
--- /dev/null
+++ b/arch/arm/boot/dts/rv1126-evb-ddr3-v13.dts
@@ -0,0 +1,31 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright (c) 2020 Rockchip Electronics Co., Ltd.
+ */
+/dts-v1/;
+#include "rv1126.dtsi"
+#include "rv11xx-evb-v12.dtsi"
+/ {
+	model = "Rockchip RV1126 EVB DDR3 V13 Board";
+	compatible = "rockchip,rv1126-evb-ddr3-v13", "rockchip,rv1126";
+
+	chosen {
+		bootargs = "earlycon=uart8250,mmio32,0xff570000 console=ttyFIQ0 root=PARTUUID=614e0000-0000 rootfstype=ext4 rootwait snd_aloop.index=7";
+	};
+
+	backlight: backlight {
+		pwms = <&pwm0 0 25000 0>;
+	};
+};
+
+&u2phy0 {
+	vup-gpios = <&gpio0 RK_PC1 GPIO_ACTIVE_LOW>;
+};
+
+&pwm0 {
+	status = "okay";
+};
+
+&pwm3 {
+	status = "disabled";
+};
-- 
2.35.3

